File: python/selection_sort.py
def selection_sort(item):
    
    for i in range(len(item)-1):

        smallest = i

        for k in range(i+1, len(item)):
            if item[k] < item[smallest]:
                smallest = k

        item[i], item[smallest] = item[smallest], item[i]

File: java_algorithms/SelectionSort.java
package algorithms.java_algorithms;

public class SelectionSort<T extends Comparable<T>> {
    
    public void sort(T[] item) {
        for (int i = 0; i < item.length - 1; i++) {

            int smallest = i;

            for (int k = i; k < item.length; k++) {

                if (item[k].compareTo(item[smallest]) < 0) {
                    smallest = k;
                }

            }

            this.swap(item, i, smallest);
        }
    }

    private void swap(T[] item, int firstIndex, int secondIndex) {
        T aux = item[firstIndex];
        item[firstIndex] = item[secondIndex];
        item[secondIndex] = aux;
    }

}
